bokeh/models/sources.py

```diff
@@ -136,6 +137,23 @@ def from_df(cls, data):
             new_data["index"] = index.tolist()
         return new_data
 
+    @classmethod
+    @deprecated("Bokeh 0.9.3")
```

Comment:
I debated doing that and chose against it because the ColumnDataSource init method has a different behavior than the from_df method did or was described to do (other than being described as a classmethod). I can add it though - it's not a big deal.